shared/chat/conversation/new-team/container.js

```diff
@@ -2,31 +2,37 @@
 import NewTeamDialog from './'
 import {connect} from 'react-redux'
 import {compose, withState, withHandlers} from 'recompose'
-import {navigateTo} from '../../../actions/route-tree'
-import {chatTab} from '../../../constants/tabs'
-import {createNewTeam, selectConversation} from '../../../actions/chat/creators'
-import {getSelectedConversation} from '../../../constants/chat'
+import {
+  createNewTeam,
+  createNewTeamFromConversation,
+  selectConversation,
+} from '../../../actions/chat/creators'
 
 import type {TypedState} from '../../../constants/reducer'
 import type {ConversationIDKey} from '../../../constants/chat'
 
-const mapStateToProps = (state: TypedState) => ({
-  conversationIDKey: getSelectedConversation(state),
+const mapStateToProps = (state: TypedState, {routeProps}) => ({
+  conversationIDKey: routeProps.conversationIDKey,
 })
 
 const mapDispatchToProps = (dispatch: Dispatch, {navigateUp}) => ({
-  navToRootChat: () => dispatch(navigateTo([], [chatTab])),
-  onBack: () => dispatch(navigateUp()),
-  onCreateTeam: (conversationIDKey: ConversationIDKey, name: string) => {
-    dispatch(createNewTeam(conversationIDKey, name))
-    dispatch(selectConversation(null, true))
+  _onCreateNewTeam: (conversationIDKey: ?ConversationIDKey, name: string) => {
+    if (conversationIDKey) {
```

Comment:
This seems hacky to me, as this component basically has to understand that if the id key is present, it's being called from the chat tab, and otherwise it's being called from the teams tab. Is there a better way to do this?

Reply:
You can pass ownProps to it to make this more explicit. Either by passing in the props in jsx or through the navigation

Reply:
I plumbed through the intended action as a property...not sure if it's the right style, though.